src/Icons/src/Iconify.php

```diff
@@ -85,6 +85,52 @@ public function fetchIcon(string $prefix, string $name): Icon
         ]);
     }
 
+    public function fetchIcons(string $prefix, array $names): array
+    {
+        if (!isset($this->sets()[$prefix])) {
+            throw new IconNotFoundException(\sprintf('The icon set "%s" does not exist on iconify.design.', $prefix));
+        }
+
+        // Sort to enforce cache hits
+        $queryString = implode(',', $names);
+        if (!preg_match('#^[\w-]+(,[\w-]+)*$#', $queryString)) {
+            throw new \InvalidArgumentException('Invalid icon names.');
+        }
+
+        // 500 chars max
+        // 42 chars: https://api.iconify.design/mdi.json?icons=
+        if (450 < \strlen($queryString)) {
+            throw new \InvalidArgumentException('The query string is too long.');
+        }
+
+        $response = $this->http->request('GET', \sprintf('/%s.json', $prefix), [
+            'headers' => [
+                'Accept' => 'application/json',
+            ],
+            'query' => [
+                'icons' => $queryString,
+            ],
+        ]);
+
+        if (200 !== $response->getStatusCode()) {
+            throw new IconNotFoundException(\sprintf('The icon set "%s" does not exist on iconify.design.', $prefix));
+        }
+
+        $data = $response->toArray();
```

Comment:
Can't we support more than 500 chars by splitting icons and doing multiple requests?

Reply:
Ah, didn't see #2352